helm/templates/dcgm-exporter-daemonset.yaml

```diff
@@ -0,0 +1,71 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: dcgm-exporter
+  namespace: {{ .Release.Namespace }}
+  labels:
+    k8s-app: dcgm-exporter
+    version: v1
+spec:
+  selector:
+    matchLabels:
+      k8s-app: dcgm-exporter
+  template:
+    metadata:
+      labels:
+        k8s-app: dcgm-exporter
+        version: v1
+    spec:
+      priorityClassName: system-node-critical
```

Comment:
I was debating on adding the priority too, but it would make sense to give dcgm-exporter a higher priority at the node level to ensure that its pod runs before the agent. This way we could avoid race condition where agent failing to fetch dcgm data if it starts before dcgm. We can still remove this config and let the scheduling to handle them normally. In the worst case scenario, agent prints warning messages or skips dcgm data until dcgm pod is ready.